front/src/reducers/osrdconf/index.ts

```diff
@@ -320,11 +320,13 @@ export function updateInfraID(infraID: any) {
     });
     dispatch(updateSwitchTypes({}));
 
-    try {
-      const newSwitchTypes = await getSwitchTypes(infraID);
-      dispatch(updateSwitchTypes(newSwitchTypes));
-    } catch (e) {
-      console.log('error on switch type retrieval', e);
+    if (infraID) {
+      try {
+        const newSwitchTypes = await getSwitchTypes(infraID);
+        dispatch(updateSwitchTypes(newSwitchTypes));
+      } catch (e) {
+        /* empty */
```

Comment:
not sure to get why we have to swallow the error here, shouldn't the log still be there or just the error be thrown if it fails ?

Reply:
Actually, I think we should change our error catching system: the error is already catch in getSwitchType() and printed in the console automatically (because if there is an error, it's a network error, the request has failed), but in the catch of getSwitchType, we throw a new error again, which is the reason why we need a new try/catch loop here.

I think we should remove the second error, but as we are going to change our call to the api  system soon (RTK), it may be not relevant to make this change now...

Reply:
okk, lets go for this then :smile: